transforms/utils/doesNotUseArguments.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+module.exports = function(j) {
+  const ReactUtils = require('./ReactUtils')(j);
+
+  const doesNotUseArguments = (path, filepath) => {
+    const hasArguments =
+      j(path)
+        .find(j.Identifier, { name: 'arguments' })
+        .size() > 0;
+
+    if (hasArguments) {
+      var warnStr = '';
+
+      if (filepath) {
+        warnStr += filepath + ': ';
+      }
+
+      warnStr += '`' + ReactUtils.directlyGetComponentName(path) + '` ' +
```

Comment:
The `directlyGetComponentName` outputs empty strings with/without these changes. Haven't dug in deep.